compiler/optimizer/VPHandlers.cpp

```diff
@@ -12214,105 +12216,15 @@ TR::Node *constrainArrayStoreChk(OMR::ValuePropagation *vp, TR::Node *node)
          if (base->getFirstChild()->getOpCode().isArrayRef())
             base = base->getFirstChild()->getFirstChild();
          }
-
-      if (base == arrayRef)
-         canBeRemoved = true;
       }
```

Comment:
How is this case (`base == arrayRef`) addressed in `OMR::ValuePropagation::isArrayStoreCheckNeeded`? `OMR::ValuePropagation::isArrayStoreCheckNeeded` only looks at `objectRef` not `objectRef->getFirstChild()->getFirstChild()`

Reply:
Thanks for catching that!  I think I should have written this at line 12223:

```
bool isStoreCheckNeeded =
        (base != arrrayRef)
        && vp->isArrayStoreCheckNeeded(arrayRef, objectRef, mustFail, storeClassForCheck, componentClassForCheck);
```

I would suggest you remove your approval

Reply:
I haven't approved the change

Reply:
> I haven't approved the change

I must have been dreaming!  :-S

Fixed in commit e516f9868d11f2483bd2f917ede6b879a1271006